File: examples/city_generator.lua
-- City Generator for OpenTibiaBR RME
if not app.hasMap() then
    return
end

app.transaction(function(map)
    local originX, originY, z = 80, 120, 7
    local width, height = 20, 14

    for dx = 0, width - 1 do
        for dy = 0, height - 1 do
            local x, y = originX + dx, originY + dy
            local tile = map:getOrCreateTile(x, y, z)
            if dx == 0 or dy == 0 or dx == width - 1 or dy == height - 1 then
                tile.ground = 1495
            else
                tile.ground = 415
            end
        end
    end

    for roadX = originX + 2, originX + width - 3 do
        local roadTile = map:getOrCreateTile(roadX, originY + 7, z)
        roadTile.ground = 393
    end

    for roofY = originY + 2, originY + height - 3, 4 do
        local wallTile = map:getOrCreateTile(originX + 4, roofY, z)
        wallTile:addItem(1497)
    end
end)

File: generators/city_generator.py
class CityGenerator:
    def __init__(self, width: int = 20, height: int = 20, depth: int = 1):
        self.width = width
        self.height = height
        self.depth = depth

    def generate_lua(self, description: str) -> str:
        return (
            "-- City generator script for RME\n"
            "if not app.hasMap() then\n"
            '  error("No hay mapa cargado en RME.")\n'
            "end\n\n"
            "app.transaction(function(map)\n"
            "  -- Crear plazas, calles y edificios principales\n"
            "  local center = { x = 100, y = 100, z = 7 }\n"
            "  local tile = map:getOrCreateTile(center.x, center.y, center.z)\n"
            "  tile:setGround(455) -- piedra pulida\n"
            "end)\n"
        )
